File: JavaScript/71-Simplify Path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    let stack = path.split("/");
    let res = [];
    for (let i=0; i<stack.length; i++){
        if (stack[i] === ".."){
            if(res.length){
                res.pop();
            }
        }else if( stack[i] && stack[i] !== "."){
            res.push(stack[i]);
        }
    }
    return "/" + res.join("/");
};

File: 71-Simplify Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = path.split("/")
        res = []
        for i in range(len(stack)):
            if stack[i] == "..":
                if res:
                    res.pop()
        
            elif stack[i] and stack[i] != ".":
                res.append(stack[i])
        
        return "/" + "/".join(res)